proxy/nginx.conf

```diff
@@ -0,0 +1,71 @@
+user  nginx;
+worker_processes  auto;
+
+error_log  /var/log/nginx/error.log warn;
+pid        /var/run/nginx.pid;
+
+events {
+    worker_connections  1024;
+}
+
+http {
+    include /etc/nginx/mime.types;
+    default_type text/plain;
+
+    map $uri:$status:$request_method $loggable_module {
+        ~(\.tar\.gz)\:200\:GET$ 1;
+        default 0;
+     }
+
+    map $uri $module {
+        ~/modules/(?<captured_module>[^/]*)(.*(\.tar\.gz)) $captured_module;
+    }
+
+    log_format json escape=json
+        '{'
+        '"time_local":"$time_local",'
+        '"module":"$module",'
+        '"upstream_addr":"$upstream_addr",'
+        '"request":"$request",'
+        '"body_bytes_sent":"$body_bytes_sent",'
+        '"request_time":"$request_time",'
+        '"http_user_agent":"$http_user_agent"'
+        '}';
+
+
+    sendfile on;
+    keepalive_timeout  65;
+
+    server {
+        listen 80;
+        server_name default_server;
+        error_page 404  /404.html;
+        set $bucket "cfbs.s3.amazonaws.com";
+
+        location / {
+            #write modules.log if file ends with .tar.gz and response code is 2xx and request method is GET
+            access_log  /home/proxy/modules.log json if=$loggable_module;
+            resolver 8.8.8.8;
```

Comment:
Why hard-coding a resolver?

Reply:
let me guess: without it, it will cache the DNS resolution forever, and fail when S3 domain maps to a different set of IP addresses.

Reply:
yes, AWS load balancer changes IP and this is needed to avoid caching the old one.

Reply:
Please see https://serverfault.com/questions/240476/how-to-force-nginx-to-resolve-dns-of-a-dynamic-hostname-everytime-when-doing-p/593003#593003